src/createrepo_c.c

```diff
@@ -1306,9 +1308,51 @@ main(int argc, char **argv)
     // Wait until pool is finished
     g_thread_pool_free(pool, FALSE, TRUE);
 
+    GHashTableIter iter;
+    gpointer key, value;
+
+    g_hash_table_iter_init(&iter, user_data.nevra_table);
+    while (g_hash_table_iter_next(&iter, &key, &value))
+    {
+        gchar *nevra = (gchar *) key;
+        GArray *locations = (GArray *) value;
+        if (locations->len > 1) {
+            g_warning("Package '%s' has duplicate metadata entries, only one should exist", nevra);
+            int strsort(const gpointer a, const gpointer b) {
+                gchar **a_ptr = (gchar **)a;
+                gchar **b_ptr = (gchar **)b;
+
+                // return strcmp(*a_ptr, *b_ptr);
+                int a_len = strlen(*a_ptr);
+                int b_len = strlen(*b_ptr);
+                if (a_len > b_len) {
+                    return 1;
+                } else if (b_len > a_len) {
+                    return -1;
+                } else {
+                    return 0;
+                }
+            }
+
+            g_array_sort(locations, strsort);
```

Comment:
```
(.env) [dalley@thinkpad test]$ createrepo_c .
Directory walk started
Directory walk done - 5 packages
Temporary output repo path: ./.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Warning: Package 'htop-0:3.2.1-1.fc35.x86_64' has duplicate metadata entries, only one should exist
Warning:     Sourced from location: 'htop-3.2.1-1.fc35.x86_64.rpm'
Warning:     Sourced from location: 'htop-3.2.1-1.fc35_rename.x86_64.rpm'
Warning: Package 'wget-0:1.21.3-1.fc35.x86_64' has duplicate metadata entries, only one should exist
Warning:     Sourced from location: 'wget-1.21.3-1.fc35.x86_64.rpm'
Warning:     Sourced from location: 'foo/wget-1.21.3-1.fc35.x86_64.rpm'
Warning:     Sourced from location: 'bar/wget-1.21.3-1.fc35.x86_64.rpm'
Pool finished with errors
```

Reply:
Just a nit, do we want to sort reverse-alphabetically?